libevmasm/ConstantOptimiser.cpp

```diff
@@ -30,10 +30,12 @@ unsigned ConstantOptimisationMethod::optimiseConstants(
 	bool _isCreation,
 	size_t _runs,
 	solidity::EVMVersion _evmVersion,
-	Assembly& _assembly,
-	AssemblyItems& _items
+	Assembly& _assembly
 )
 {
+	// TODO: design the optimiser in a way this is not needed
+	AssemblyItems& _items = const_cast<AssemblyItems &>(_assembly.items());
```

Comment:
Ah ok, if you also need a const_cast here, then the other version is of course better. Sorry, I though the const_cast would only be needed for the tests.

Reply:
Changed again, though this time using `items` and not `itemsMutable`. Let me know which one is preferred.